src/Translumo.TTS/TtsConfiguration.cs

```diff
@@ -1,22 +1,32 @@
 ﻿using Translumo.Infrastructure.Language;
 using Translumo.Utils;
+using Windows.Security.EnterpriseData;
 
 namespace Translumo.TTS;
 
 
 public class TtsConfiguration : BindableBase
 {
-    public static TtsConfiguration Default =>
-        new TtsConfiguration()
-        {   
-            TtsLanguage = Languages.English,
-            TtsSystem = TTSEngines.None,
-            InstalledWinTtsLanguages = new List<Languages>()
-        };
+    protected TtsConfiguration()
+    {
+        // for serialization
+    }
+
+    public TtsConfiguration(LanguageService languageService)
+    {
+        TtsLanguage = Languages.English;
+        TtsSystem = TTSEngines.None;
+        InstalledWinTtsLanguages = new List<Languages>();
+        _languageService = languageService;
+    }
 
     private TTSEngines _ttsSystem;
     private Languages _ttsLanguage;
     private List<Languages> _installedWinTtsLanguages;
+    private readonly LanguageService _languageService;
+
+    public bool IsLanguageSupportedInTtsEngine(TTSEngines engine, Languages lang) =>
```

Comment:
In truth, I don't really like that dependency of TtsConfiguration on TtsFactory (and also it brings Languageservice dependency). I suggest just use non-static TtsFactory method directly in VM.

Reply:
removed after moving to `CreateSileroTtsCheckingStages`